R/MsBackendMetaboLights.R

```diff
@@ -0,0 +1,267 @@
+#' @title MsBackend representing MS data from MetaboLights
+#'
+#' @name MsBackendMetaboLights
+#'
+#' @aliases MsBackendMetaboLights-class
+#'
+#' @description
+#'
+#' The `MsBackendMetaboLights` retrieves and represents mass spectrometry (MS)
+#' data from metabolomics experiments stored in the
+#' [MetaboLights](https://www.ebi.ac.uk/metabolights/) repository. The backend
+#' directly extends the [MsBackendMzR] backend from the *Spectra* package and
+#' hence supports MS data in mzML, netCDF and mzXML format. Upon initialization
+#' with the `backendInitialize()` method, the `MsBackendMetaboLights` backend
+#' downloads and caches the MS data files of an experiment locally avoiding
+#' hence repeated download of the data.
+#'
+#' @section Initialization and loading of data:
+#'
+#' New instances of the class can be created with the `MsBackendMetaboLights()`
+#' function. Data is loaded and initialized using the `backendInitialize()`
+#' function with parameters `mtblsId`, `assayId` and `filePattern`. `mtblsId`
+#' must be the ID of a **single** (existing) MetaboLights data set. Parameter
+#' `assayId` allows to define specific *assays* of the MetaboLights data set
+#' from which the data files should be loaded. If provided, it should be the
+#' file names of the respective assays in MetaboLights (use e.g.
+#' `mtbls_list_files(<MetaboLights ID>, pattern = "^a_")` to list all available
+#' assay files for a given MetaboLights ID `<MetaboLights ID>`. By default,
+#' with `assayId = character()` MS data files from all assays of a data set
+#' are loaded. Optional parameter `filePattern` defines the pattern that should
```

Comment:
is the parameter  `filePattern`  necessary ? the user just want to download what's available for the assay they want no ?

Reply:
ok so I read the part about this in the vignette, i think i understand it's give freedom to the user to select more specifically which files they want.. We agree that all of these possibilities also need to be available to the user through the MetabolightsParam() when loading the MsExperiment() object?

Reply:
The *standard* use case would indeed be that users load all files of one data set. The `filePattern` was initially to define the file types that are supported/can be read. But it can also be used as described in the vignette. The *main* use case would be more in cases where the file endings are not matching the default ones (e.g. mzml instead of mzML).

I would maybe use the `filePattern` parameter also in `MsExperiment()`. for the `assayId`, I would also add that parameter, just to allow manual definition of a single assay name if needed. The interactive selection/choice in case of multiple assays is great, but it would not work e.g. in a Rmd file - only when running interactively. Means, I would keep you choice selection you have in *MsIO*, but bypass that if the user manually specifies a single assay.

Reply:
ah damn ok i did not think of the mzml vs mzML example ! make sense thanks, and i completely agree on the assayId parameter comment.